### PR TITLE
Avoid printing erroneous warning about "MIOpen not found" for ROCm builds

### PR DESCRIPTION
**Pull request by jithunnair-amd**
Older versions of MIOpen (<=2.2) don't have the `miopenGetVersion` api, but MIOpen is always a part of the ROCm builds, so do NOT set `lib` to None for ROCm builds. `__cudnn_version` will be `None` for older versions of MIOpen.

Setting `lib` to `None` ends up printing the following erroneous warning when running unit tests:
```
/root/.local/lib/python3.6/site-packages/torch/backends/cudnn/__init__.py:120: UserWarning: cuDNN/MIOpen library not found. Check your LD_LIBRARY_PATH
  }.get(sys.platform, 'LD_LIBRARY_PATH')))
```
Eg.: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py3.6-clang7-rocmdeb-ubuntu16.04-test2/18387/consoleFull

### CODE REVIEW

**Comment by t-vi** [comment]
The ROCm test failures seem legit: Now PyTorch wants to run CuDNN RNN weight flattening on MIOpen, which fails.

**Comment by jithunnair-amd** [comment]
Yes, the failures are legit. I'm working with some other team members internally to see how we can fix these unit tests.

**Comment by t-vi** [comment]
You could either special case ROCm based on version.hip_version or -- similar to what we did for CTC loss -- provide a function to query [whether the cudnn function is available](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/LossCTC.cpp#L15) (ok, this one combines "is applicable" with "is available", but I think it's not unreasonable). This should be an internal function (starting with `_`).

**Comment by jithunnair-amd** [comment]
Since PR #34265 is merged...

@pytorchbot retest this please

File: torch/testing/_internal/common_cuda.py
Comment (xw285cornell) [inline torch/testing/_internal/common_cuda.py]: dumb question: why we no longer need TEST_WITH_ROCM here?

Comment (jithunnair-amd) [reply torch/testing/_internal/common_cuda.py]: The `is_acceptable` function would previously return `False` for ROCm builds, so we would override it by using `TEST_WITH_ROCM`, but with my changes, the function should return `True` even for ROCm builds, rendering the workaround unnecessary.

**Review by xw285cornell — COMMENTED** [review]
LGTM overall! I have some infra flakeness in the internal build. Since OSS build all passed I think it should be good to go, will get back in merging it when our internal signal is clean.

**Review by xw285cornell — CHANGES_REQUESTED** [review]
Sorry, this breaks our internal build because torch.version doesn't have hip. I'll try to make a fix (not visible to you) and try again. 

I'll block this PR first to avoid it gets landing (it's breaking everything in our internal build). Nothing needs to be done on your side.

**Comment by jithunnair-amd** [comment]
Okay, thank you. I see some conflicts in `torch/backends/cudnn/__init__.py` which I think are arising from the PR https://github.com/pytorch/pytorch/pull/33678.

**Comment by xw285cornell** [comment]
With that one already landed, do you still need this PR?

**Comment by jithunnair-amd** [comment]
I'll reassess and get back to you.

**Comment by jithunnair-amd** [comment]
@xw285cornell I removed the changes to `torch/backends/cudnn/__init__.py` since those were not needed anymore. However, there was a leftover ROCm-specific oversight in  https://github.com/pytorch/pytorch/pull/33678#discussion_r391851587 which is addressed in https://github.com/pytorch/pytorch/pull/33837/commits/734dc8af96be64e1d2d493e8b34038647ebc0d36.

**Comment by jithunnair-amd** [comment]
@xw285cornell I'd request that we don't merge this until the ROCm CI runs finish and we see that the warnings have disappeared (builds won't fail since it's only a warning). I'll let you know once I've checked the logs.

**Comment by xw285cornell** [comment]
Yes, I was just importing this PR into our internal CI system to make sure nothing breaks. I'll wait for all green lights before I merge it.

**Comment by jithunnair-amd** [comment]
@xw285cornell The warning doesn't appear anymore with the changes in this PR. Ready to merge.

**Review by xw285cornell — APPROVED** [review]

**Pull request closed without merging**
